bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -0,0 +1,220 @@
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_INTENSITY, PARAM_PRECIPITATION_1H, MILLIMETRE_PER_HOUR);
+        addMapping(CHANNEL_TOTAL_CLOUD_COVER, PARAM_TOTAL_CLOUD_COVER, PERCENT);
+        addMapping(CHANNEL_FORECAST_WEATHER_ID, PARAM_WEATHER_SYMBOL, null);
+    }
+
+    private @NonNullByDefault({}) LatLon location;
+
+    public ForecastWeatherHandler(Thing thing) {
+        super(thing);
+        // Override poll interval to slower value
+        pollIntervalSeconds = (int) TimeUnit.MINUTES.toSeconds(QUERY_RESOLUTION_MINUTES);
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            Object location = getConfig().get(BindingConstants.LOCATION);
+            if (location == null) {
+                throw new IllegalStateException();
+            }
+            String latlon = location.toString();
+            String[] split = latlon.split(",");
+            this.location = null;
+            this.location = new LatLon(new BigDecimal(split[0].trim()), new BigDecimal(split[1].trim()));
+            super.initialize();
+        } catch (IllegalStateException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter not set %s %s", e.getClass().getName(), e.getMessage()));
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter should be in format LATITUDE,LONGITUDE. Error: %s %s",
+                            e.getClass().getName(), e.getMessage()));
+        }
+    }
+
+    @Override
+    protected void update() {
+        if (location == null) {
+            return;
+        }
+        for (int retry = 0; retry < RETRIES; retry++) {
+            try {
+                long now = Instant.now().getEpochSecond();
+                response = client.query(new ForecastRequest(location, floorToEvenMinutes(now, QUERY_RESOLUTION_MINUTES),
+                        ceilToEvenMinutes(now + TimeUnit.HOURS.toSeconds(FORECAST_HORIZON_HOURS),
+                                QUERY_RESOLUTION_MINUTES),
+                        QUERY_RESOLUTION_MINUTES), TIMEOUT_MILLIS);
+            } catch (FMIResponseException e) {
+                response = null;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        String.format("%s: %s", e.getClass().getSimpleName(), e.getMessage()));
+                try {
+                    Thread.sleep(RETRY_DELAY_MILLIS);
```

Comment:
I'm not sure how that would work out really since this method is called with `scheduleWithFixedDelay`.

Assuming exceptions are rare, I find this current approach acceptable.

Reply:
If I see correctly, the scheduler's thread pool is shared amongst all ThingHandlers in the JVM.

I would propose to remove the for-loop and in case of an exception, schedule a one-shot `update()` again with the retry timeout. The retry counter increments, if a one-shot timer is scheduled. I think is can be easily done.

Reply:
Thanks for checking, then it certainly makes sense to optimize.

Reply:
Implemented in e455d438856140cd7f0d3435f5165e59af3ddec3 , appreciate second pair of eyes